### PR TITLE
Create CustomPopUps.

### PR DESCRIPTION
**Pull request by tamokakhidze**
Create CustomPopUp for logging out and closing question. Create PopUpButton to use on custom pop ups. Create FinalScorePopUp which is used when you finish quiz.

### CODE REVIEW

File: QuizApp/Customs/CustomPopUp.swift [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+//  CustomPopUp.swift
+//  QuizApp
+//
+//  Created by Tamuna Kakhidze on 18.11.24.
+//
+
+import UIKit
+
+// MARK: - CustomPopUp
+class CustomPopUp: UIView {
```
Comment (LukaGujejiani) [inline QuizApp/Customs/CustomPopUp.swift]: Make this class final

File: QuizApp/Customs/FinalScorePopUp.swift [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+//  FinalScorePopUp.swift
+//  QuizApp
+//
+//  Created by Tamuna Kakhidze on 18.11.24.
+//
+
+import UIKit
+
+// MARK: - FinalScorePopUp
+class FinalScorePopUp: UIView {
```
Comment (LukaGujejiani) [inline QuizApp/Customs/FinalScorePopUp.swift]: Final class

**Review by LukaGujejiani — APPROVED** [review]

File: QuizApp/Customs/CustomPopUp.swift (hunk 1 above)
Comment (AnaIoramashvili) [inline QuizApp/Customs/CustomPopUp.swift]: add keyword "final"

File: QuizApp/Customs/FinalScorePopUp.swift (hunk 2 above)
Comment (AnaIoramashvili) [inline QuizApp/Customs/FinalScorePopUp.swift]: add keyword "final"

**Review by AnaIoramashvili — CHANGES_REQUESTED** [review]

File: QuizApp/Customs/CustomPopUp.swift (hunk 1 above)
Comment (AnaIoramashvili) [reply QuizApp/Customs/CustomPopUp.swift]: Sorry, just saw that Luka wrote the same comment

**Review by AnaIoramashvili — APPROVED** [review]

File: QuizApp/Customs/CustomPopUp.swift [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+// MARK: - CustomPopUp
+final class CustomPopUp: UIView {
+    
+    // MARK: - UI Components
+    private let mainStackView: UIStackView = {
+        let stackView = UIStackView()
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.axis = .vertical
+        stackView.alignment = .fill
```
Comment (AniDzamelashvili) [inline QuizApp/Customs/CustomPopUp.swift]:
```suggestion
```

Comment (AniDzamelashvili) [reply QuizApp/Customs/CustomPopUp.swift]: It's default value for alignment

File: QuizApp/Customs/CustomPopUp.swift [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+        stackView.distribution = .fillEqually
+        stackView.spacing = Constants.Sizing.mainStackViewSpacing
+        return stackView
+    }()
+    
+    private let buttonsStackView: UIStackView = {
+        let stackView = UIStackView()
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.axis = .horizontal
```
Comment (AniDzamelashvili) [inline QuizApp/Customs/CustomPopUp.swift]:
```suggestion
```

Comment (AniDzamelashvili) [reply QuizApp/Customs/CustomPopUp.swift]: It's default value for axis

File: QuizApp/Customs/PopupButton.swift [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    private func initialize() {
+        setAppearance()
+        setConstraints()
+    }
+    
+    private func setAppearance() {
+        self.backgroundColor = Constants.Colors.backgroundColor
+        self.clipsToBounds = true
+        self.contentHorizontalAlignment = .center
+        self.contentVerticalAlignment = .center
```
Comment (AniDzamelashvili) [inline QuizApp/Customs/PopupButton.swift]: I think contentHorizontalAlignment is by default centre, but not sure. please check for contentVerticalAlignment as well

**Review by AniDzamelashvili — COMMENTED** [review]
Good job🚀

**Review by AniDzamelashvili — APPROVED** [review]

**Pull request merged**